### PR TITLE
fix(sync)_: Improve EnableInstallationAndSync to include installation in MessengerResponse

### PR DESCRIPTION
**Pull request by qfrank**
Major changes:
- Refactor EnableInstallationAndSync for better error handling and response merging
- Add new EnableInstallationV2 method returning the installation
- Update tests to check for installation in response
- Deprecate old EnableInstallation method

fix relate [comment](https://github.com/status-im/status-mobile/issues/21259#issuecomment-2367648024)

status: ready

### CODE REVIEW

**Review by jrainville — APPROVED** [review]
Looks good. As far as I can see, this doesn't modify the API so no work is needed on the client side right?

**Comment by qfrank** [comment]
> Looks good. As far as I can see, this doesn't modify the API so no work is needed on the client side right?

not sure if desktop exists the same [issue](https://github.com/status-im/status-mobile/issues/21259) as mobile, if desktop does not need updated installation info within MessengerResponse returned from EnableInstallationAndSync, that's okay. @jrainville

File: protocol/messenger_pairing_and_syncing.go [hunk 1]
```diff
@@ -512,6 +520,23 @@ func (m *Messenger) EnableInstallation(id string) error {
 	return nil
 }
 
+// EnableInstallationV2 enables an installation and returns the installation
+func (m *Messenger) EnableInstallationV2(id string) (*multidevice.Installation, error) {
```
Comment (ilmotta) [inline protocol/messenger_pairing_and_syncing.go]: The V2 function seems to be identical, except for an extra return value. If that's correct, then I don't see the need to create a new function and deprecate the old one. We can just change the original function to return an extra value and change all callers that don't care about the `installation` to ignore with an underscore. This refactor would be guaranteed by the compiler, so no risk of regression, and we wouldn't create unnecessary tech debt.

Comment (qfrank) [reply protocol/messenger_pairing_and_syncing.go]: fixed in latest commit @ilmotta

**Review by ilmotta — APPROVED** [review]

**Comment by qfrank** [comment]
close this and use [PR 5888](https://github.com/status-im/status-go/pull/5888) instead cc @Parveshdhull

**Pull request closed without merging**